drivers/smbus/smbus_utils.h

```diff
@@ -107,4 +110,82 @@ static inline void smbus_init_callback(struct smbus_callback *callback,
  */
 void smbus_loop_alert_devices(const struct device *dev, sys_slist_t *callbacks);
 
+/**
+ * @brief Compute the number of messages required for an SMBus transaction
+ *
+ * If @p flags indicates that the transaction requires packet error checking (PEC),
+ * the number of messages is equal to @p num_msgs, otherwise the number of messages
+ * required is @p num_msgs - 1, since a PEC byte is not required.
+ *
+ * Callers are expected to allocated an array of @ref i2c_msg objects to hold a number of
+ * i2c messages, including one message dedicated to the a PEC byte, whether or not PEC is
+ * being used.
+ *
+ * @note This function is intended for SMBus drivers that do not have hardware PEC support and
+ * therefore must rely on software PEC calculation.
+ *
+ * @param flags SMBus flags.
+ * @param num_msgs Number of allocated messages.
+ * @return The number of required messages.
+ */
+uint8_t smbus_pec_num_msgs(uint32_t flags, uint8_t num_msgs);
+
+/**
+ * @brief Compute the packet error checking (PEC) byte for an SMBus transaction
+ *
+ * @note This function is intended for SMBus drivers that do not have hardware PEC support and
+ * therefore must rely on software PEC calculation.
+ *
+ * @note At this time, only 7-bit addresses are supported in @p addr.
+ *
+ * @param addr The address of the target device.
+ * @param msgs Array of @ref i2c_msg that make up the transaction.
+ * @param num_msgs Number of messages in the transaction.
+ * @return the computed PEC byte.
+ */
+uint8_t smbus_pec(uint16_t addr, const struct i2c_msg *msgs, uint8_t num_msgs);
+
+/**
+ * @brief Prepare the packet error checking (PEC) byte for an SMBus write transaction
+ *
+ * If the @p flags bitmask contains @ref SMBUS_MODE_PEC (i.e. PEC is enabled), the number of
+ * messages is equal to @p num_msgs, otherwise the number of messages required is @p num_msgs - 1,
+ * since a PEC byte is not required.
+ *
+ * @note This function is intended for SMBus drivers that do not have hardware PEC support and
+ * therefore must rely on software PEC calculation.
+ *
+ * @note At this time, only 7-bit addresses are supported in @p addr.
+ *
+ * @param flags SMBus flags.
+ * @param addr The address of the target device.
+ * @param messages Array of @ref i2c_msg objects that make up the transaction.
```

Comment:
```suggestion
 * @param msgs Array of @ref i2c_msg objects that make up the transaction.
```
Ditto for `smbus_read_check_pec()`.